### PR TITLE
Centralize WPILib Contributing Articles

### PR DESCRIPTION
**Pull request by Daltz333**
- [x] WPILib Guidelines
- [x] WPILib Building
- [ ] WPILib CMake Building
- [ ] WPILib Faster Builds
- [ ] WPILib Styleguide Java/C++
- [ ] WPILib Pyformat

### CODE REVIEW

File: source/docs/contributing/allwpilib/building.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+   ./gradlew :wpilibc:build
+
+If you have installed the FRC Toolchain to a directory other than the default, or if the Toolchain location is not on your System PATH, you can pass the ``toolChainPath`` property to specify where it is located. Example:
+
+.. code-block:: shell
+
+   ./gradlew build -PtoolChainPath=some/path/to/frc/toolchain/bin
+
+If you also want simulation to be built, add -PmakeSim. This requires gazebo_transport. We have tested on 14.04 and 15.05, but any correct install of Gazebo should work, even on Windows if you build Gazebo from source. Correct means CMake needs to be able to find gazebo-config.cmake. See `The Gazebo website <http://gazebosim.org/>`__ for installation instructions.
```
Comment (prateekma) [inline source/docs/contributing/allwpilib/building.rst:45]: Use code formatting around `-PmakeSim`.

File: source/docs/contributing/allwpilib/building.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+.. code-block:: shell
+
+   ./gradlew build -PmakeSim
+
+If you prefer to use CMake directly, the you can still do so. The common CMake tasks are wpilibcSim, frc_gazebo_plugins, and gz_msgs
```
Comment (prateekma) [inline source/docs/contributing/allwpilib/building.rst:51]: Remove "the" after the comma.

File: source/docs/contributing/allwpilib/contribution-guidelines.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Merge Process
+^^^^^^^^^^^^^
+
+When you first submit changes, Travis-CI will attempt to run ``./gradlew check`` on your change. If this fails, you will need to fix any issues that it sees. Once Travis passes, we will begin the review process in more earnest. One or more WPILib team members will review your change. This will be a back-and-forth process with the WPILib team and the greater community. Once we are satisfied that your change is ready, we will allow our Jenkins instance to test it. This will run the full gamut of checks, including integration tests on actual hardware. Once all tests have passed and the team is satisfied, we will merge your change into the WPILib repository.
```
Comment (prateekma) [inline source/docs/contributing/allwpilib/contribution-guidelines.rst:56]: This should probably be updated with the latest info -- Travis isn't used anymore and it's all Azure.

File: source/docs/contributing/allwpilib/building.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+===============
+
+Requirements
+------------
+
+- A C++ compiler
+
+  - On Linux, GCC 7+ works fine
+  - On Windows, you need Visual Studio 2019 (the free community edition works fine). Make sure to select the C++ Programming Language for installation
+
```
Comment (AustinShalit) [inline source/docs/contributing/allwpilib/building.rst:11]:
```suggestion
  - On macOS, you need to have Xcode installed 

```

**Comment by Daltz333** [comment]
If anyone else wants to take this on, help is welcome.

**Pull request closed without merging**